src/pypi_changes/_print.py

```diff
@@ -32,23 +33,55 @@ def print_tree(distributions: Iterable[Package], options: Options) -> None:
         text.stylize(f"link https://pypi.org/project/{pkg.name}/#history")
         text.append(" ", "white")
         text.append(pkg.version, "blue")
-        last_release = pkg.last_release
-        if last_release is not None:  # pragma: no branch
-            if last_release["version"] != pkg.version and pkg.info is not None:
-                for a_version, releases in pkg.info["releases"].items():  # pragma: no branch
-                    first_release_at = releases[0]["upload_time_iso_8601"]
-                    if a_version == pkg.dist.version and first_release_at is not None:
-                        text.append(" ")
-                        text.append(naturaltime(now - first_release_at), "green")
-                        break
-                text.append(f" remote {last_release['version']}", "red")
-            if last_release["upload_time_iso_8601"] is not None:  # pragma: no branch
+
+        current_release = pkg.current_release
+        current_release_at = current_release.get("upload_time_iso_8601")
+        last_release = pkg.last_release or {}
+        last_release_at = pkg.last_release_at
+
+        if current_release_at is not None:
+            text.append(" ")
+            text.append(naturaltime(now - current_release_at), "green")
+        if pkg.version != last_release.get("version"):
+            text.append(f" remote {last_release.get('version')}", "red")
+            if last_release_at is not None:  # pragma: no branch
                 text.append(" ", "white")
-                text.append(naturaltime(now - last_release["upload_time_iso_8601"]), "green")
+                text.append(naturaltime(now - last_release_at), "green")
         tree.add(text)
     rich_print(tree)
 
 
+def release_info(release: dict[str, Any] | None, now: datetime) -> dict[str, Any]:
```

Comment:
I'd prefer to make the _print a package that has a two modules rich and json. And each file can only contain it's own logic. Can you do that please?

Reply:
Sure.

Reply:
I've done the split into `_print.json` and `_print.rich`. Not sure about what naming conventions you want to follow for these modules.